src/mui/input/ReferenceInput.js

```diff
@@ -157,28 +160,37 @@ export class ReferenceInput extends Component {
             resource,
             label,
             source,
-            referenceRecord,
             allowEmpty,
             matchingReferences,
             basePath,
             onChange,
             children,
             meta,
+            translate,
         } = this.props;
-        if (!referenceRecord && !allowEmpty) {
```

Comment:
Aren't we still missing that case ?

Reply:
What we should do: 

- Display an error message saying that the reference wasn't found in the API
- display the input with an empty value. It should be functional (i.e. allow to set a new value)

Reply:
The error shouldn't be displayed by the `ReferenceInput` itself right ? It should just be passed to the child as `ReferenceInput` don't actually render anything.

Reply:
You're right